### ISSUE TITLE
Name transactions not recognized in descriptor wallets

### ISSUE THREAD

**Issue by JeremyRand** [namecoin/namecoin-core#429]
Descriptor wallets don't seem to recognize name transactions.

**Expected behavior**

`listtransactions` and `name_list` should show a name transaction.

**Actual behavior**

`listtransactions` and `name_list` yield `[]`

**To reproduce**

1. Register a name in a non-descriptor wallet.
2. Create a descriptor wallet.
3. Create a receive address in the descriptor wallet.
4. Transfer the name to the descriptor wallet's receive address via `name_update` in the non-descriptor wallet.
5. Run `listtransactions` and `name_list` against the descriptor wallet.

**System information**

Namecoin Core 67a184aa77c8c3ff439bfdbac414e67057445dd9, self-compiled.

Whonix 15, ppc64le, spinning-rust HDD.

Interestingly, the `name_show` method *does* show `"ismine": true` when run against the descriptor wallet.

Nothing obvious in `debug.log` at first glance.

**Comment by JeremyRand** [comment]
Side note: I accidentally discovered this when I just transferred a name to a new descriptor wallet.  The name is not critical, but I'd also prefer not to lose it -- so it would be great if this can be fixed before 36 kiloblocks elapse from now.  :)

**Comment by domob1812** [comment]
Will look at it.  As for your name, I guess you can just do a manual raw transaction to get it back out?

**Comment by JeremyRand** [comment]
> As for your name, I guess you can just do a manual raw transaction to get it back out?

Probably easier to export the private key from the descriptor wallet, and import that into Electrum-NMC or something to get it out.  I'd assume this'll be fixed long before I need to do that though.

**Comment by JeremyRand** [comment]
Honestly I'm not surprised that descriptor wallets are broken.  AIUI, the whole point of descriptor wallets is that they match against script templates.  So it seems natural that some patching would be needed to make the matching code strip the name script prefix before trying to match the script.

**Comment by domob1812** [comment]
Yes true.  But yeah, I'll try to get to this rather sooner than later, so as long as you keep that name in the back of your mind, you should be fine.
